src/ui.rs

```diff
@@ -194,6 +271,17 @@ impl Ui {
     pub fn flush(&mut self) -> io::Result<()> {
         match &mut self.output {
             UiOutput::Terminal { stdout, .. } => stdout.flush(),
+            UiOutput::Paged { child } => child.stdin.take().unwrap().flush(),
+        }
+    }
+
+    pub fn finalize_writes(&mut self) -> io::Result<()> {
+        match &mut self.output {
+            UiOutput::Paged { child } => {
+                child.wait()?;
+                Ok(())
```

Comment:
Do you think we should restore the output to use the regular stdout and stderr here? I ask because I saw that `main.rs` and the example applications call this just before handling errors. Maybe we would ideally print the errors to stderr instead of printing them to the pager? Especially if the error comes from starting the pager.

Reply:
Yeah that's a good point. Let me try that.

Reply:
You can use `std::mem::replace()` to restore stdout/stderr and take ownership of `Paged { ... }`.
That's probably needed if you made `UiOutput::Paged` hold `{ child, stdin }` pair.

https://doc.rust-lang.org/std/mem/fn.replace.html

```rust
match mem::replace(&mut self.output, UiOutput::Terminal { ... }) {
    UiOutput::Paged { ... } => {
        drop(stdin); // close()
        child.wait()?;
        ...
    }
}
```

Reply:
Between this and the other comment on printing an error when we fail to spawn the pager, I think it might be better to always hold a handle to stdout/stderr instead of swapping out `Ui.output` when the pager is requested.